test/cctest/test_node_api.cc

```diff
@@ -7,35 +7,243 @@
 #include "node_binding.h"
 #include "node_test_fixture.h"
 
-using v8::Local;
-using v8::Object;
 
-static napi_env addon_env;
-
-class NodeApiTest : public EnvironmentTestFixture {
+class NodeApiTestEnv : public NodeTestFixture {
```

Comment:
The cctest for node-api is primarily used for internal behavior testing. That is to say, the behavior is not observable through node-api c interface.

All public node-api c interface should be tested in `test/node-api` if it is node specific node-api or `test/js-native-api` if the node-api is a bridge for JavaScript operations.

I believe promise related node-api testing should be located in `test/js-native-api`.

Reply:
ok,thank you。